service/src/main/java/com/fecklessweasel/service/FileDownloadServlet.java

```diff
@@ -0,0 +1,80 @@
+package com.fecklessweasel.service;
+
+import com.fecklessweasel.service.datatier.SQLInteractionInterface;
+import com.fecklessweasel.service.datatier.SQLSource;
+import com.fecklessweasel.service.objectmodel.OMUtil;
+import com.fecklessweasel.service.objectmodel.ServiceException;
+import com.fecklessweasel.service.objectmodel.StoredFile;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+/**
+ * Downloads the given file for the user.
+ * @author James Flinn
+ */
+@WebServlet("/servlet/file/download")
+public class FileDownloadServlet extends HttpServlet {
+
+    private static final String FILEPATH_PREFIX = "files";
+
+    /**
+     * When this servlet receives a GET request, downloads the given file.
+     * @param request The web request.
+     * @param response The web response.
+     */
+    @Override
+    protected void doGet(final HttpServletRequest request,
+                         HttpServletResponse response)
+            throws ServletException, IOException {
+
+        final int fid = Integer.parseInt(request.getParameter("fid"));
```

Comment:
Use OMUtil.parseInt() instead of Integer.parseInt(). It will both null check the parameter and make sure it is a number. Otherwise, if someone goes to the page and specifies no fid or a string for the fid the page will break.